### PR TITLE
feat(pci.data-processing): submitJob - add TTL and change config display

### PR DESCRIPTION
**Pull request by jperchoc**
DATATR-16

Signed-off-by: Jonathan Perchoc <jonathan.perchoc@ovhcloud.com>

| Question         | Answer
| ---------------- | ---
| Branch?          | `feat/datatr-15-jobs-sizing`
| Bug fix?         | no
| New feature?     | yes
| Breaking change? | no
| Tickets          | Fix #DATATR-16
| License          | BSD 3-Clause

- [x] Try to keep pull requests small so they can be easily reviewed.
- [x] Commits are signed-off
- [x] Only FR translations have been updated
- [x] Branch is up-to-date with target branch
- [x] Lint has passed locally
- [ ] Standalone app was ran and tested locally
- [x] Ticket reference is mentioned in linked commits (internal only)
- [ ] Breaking change is mentioned in relevant commits

## Description

Improve configuration step

## Related

```
DATATR-2 <-- DATATR-14 <-- DATATR-15 <-- DATATR-16
```
[DATATR-2](https://github.com/ovh/manager/tree/feat/datatr-2-dataprocessing-new-job)
[DATATR-14](https://github.com/ovh/manager/pull/7047)
[DATATR-15](https://github.com/ovh/manager/pull/7067)
[DATATR-16](https://github.com/ovh/manager/pull/7069)

### CODE REVIEW

File: packages/manager/modules/pci/src/projects/project/data-processing/submit-job/spark-config/spark-config.controller.js [hunk 1]
```diff
@@ -50,6 +55,30 @@ export default class {
...
 
+    this.units = [
+      {
+        name: 'minutes',
+        description: this.$translate.instant(
+          'data_processing_submit_job_stepper_spark_select_unit_minutes',
+        ),
+      },
+      {
+        name: 'hours',
```
Comment (Dimfacion) [inline packages/manager/modules/pci/src/projects/project/data-processing/submit-job/spark-config/spark-config.controller.js:66]: Aren't you missing multipliers here ?

Comment (jperchoc) [reply packages/manager/modules/pci/src/projects/project/data-processing/submit-job/spark-config/spark-config.controller.js:66]: Added: https://github.com/ovh/manager/pull/7069/commits/672e694c8562d0c4de36b0008c6ac93affa8a369

File: packages/manager/modules/pci/src/projects/project/data-processing/submit-job/spark-config/spark-config.controller.js [hunk 2]
```diff
@@ -77,6 +107,22 @@ export default class {
...
+          PYTHON_FILES_CONTENT_TYPES.includes(object.contentType),
+        );
+        const javaFiles = container.objects.filter((object) =>
+          JAVA_FILES_CONTENT_TYPES.includes(object.contentType),
+        );
+        if (pythonFiles.length > 0) {
+          this.state.jobType = 'python';
+          this.state.mainApplicationCode = pythonFiles[0].name;
+        } else if (javaFiles.length > 0) {
+          this.state.jobType = 'java';
```
Comment (JeremyDec) [inline packages/manager/modules/pci/src/projects/project/data-processing/submit-job/spark-config/spark-config.controller.js:121]: why not using `JOB_TYPE` constant here?

File: packages/manager/modules/pci/src/projects/project/data-processing/submit-job/spark-config/translations/Messages_fr_FR.json [hunk 3]
```diff
@@ -6,7 +6,7 @@
   "data_processing_submit_job_stepper_spark_jobtype_label": "Type de job",
   "data_processing_submit_job_stepper_spark_jobtype_placeholder": "Sélectionnez un type de job",
   "data_processing_submit_job_stepper_spark_jar_file_label": "Fichier JAR",
-  "data_processing_submit_job_stepper_spark_python_file_label": "Fichier Pyhon",
+  "data_processing_submit_job_stepper_spark_python_file_label": "Fichier Python",
```
Comment (JeremyDec) [inline packages/manager/modules/pci/src/projects/project/data-processing/submit-job/spark-config/translations/Messages_fr_FR.json:9]: :+1:

File: packages/manager/modules/pci/src/projects/project/data-processing/submit-job/spark-config/spark-config.controller.js (hunk 2 above)
Comment (jperchoc) [reply packages/manager/modules/pci/src/projects/project/data-processing/submit-job/spark-config/spark-config.controller.js]: Done here: https://github.com/ovh/manager/pull/7069/commits/67f807ec48058a5f5652d27e617e21166749fd46

File: packages/manager/modules/pci/src/projects/project/data-processing/submit-job/spark-config/spark-config.constants.js [hunk 4]
```diff
@@ -1,7 +1,15 @@
 export const ARGUMENTS_VALIDATION_PATTERN = /^[\w].*/;
 export const PYTHON_ENV_FILENAME = 'environment.yml';
+export const PYTHON_FILES_CONTENT_TYPES = ['text/x-python'];
```
Comment (y4nnL) [inline packages/manager/modules/pci/src/projects/project/data-processing/submit-job/spark-config/spark-config.constants.js:3]:
```suggestion
export const PYTHON_FILES_CONTENT_TYPE = 'text/x-python';
```

File: packages/manager/modules/pci/src/projects/project/data-processing/submit-job/spark-config/spark-config.controller.js [hunk 5]
```diff
@@ -77,6 +107,22 @@ export default class {
       .listObjects(this.projectId, containerId)
       .then((container) => {
         this.containerObjects = container.objects;
+        // try to detect correct job type and file
+        const pythonFiles = container.objects.filter((object) =>
+          PYTHON_FILES_CONTENT_TYPES.includes(object.contentType),
```
Comment (y4nnL) [inline packages/manager/modules/pci/src/projects/project/data-processing/submit-job/spark-config/spark-config.controller.js:112]:
```suggestion
          object.contentType === PYTHON_FILES_CONTENT_TYPE,
```

File: packages/manager/modules/pci/src/projects/project/data-processing/submit-job/spark-config/spark-config.controller.js [hunk 6]
```diff
@@ -77,6 +107,22 @@ export default class {
...
+        );
+        const javaFiles = container.objects.filter((object) =>
+          JAVA_FILES_CONTENT_TYPES.includes(object.contentType),
+        );
+        if (pythonFiles.length > 0) {
+          this.state.jobType = JOB_TYPE_PYTHON;
+          this.state.mainApplicationCode = pythonFiles[0].name;
```
Comment (y4nnL) [inline packages/manager/modules/pci/src/projects/project/data-processing/submit-job/spark-config/spark-config.controller.js:119]: **question**: what if the python files have different `name` value ?

File: packages/manager/modules/pci/src/projects/project/data-processing/submit-job/spark-config/spark-config.controller.js [hunk 7]
```diff
@@ -77,6 +107,22 @@ export default class {
...
+        } else if (javaFiles.length > 0) {
+          this.state.jobType = JOB_TYPE_JAVA;
+          this.state.mainApplicationCode = javaFiles[0].name;
```
Comment (y4nnL) [inline packages/manager/modules/pci/src/projects/project/data-processing/submit-job/spark-config/spark-config.controller.js:122]: **question**: what if the java files have different `name` value ?

File: packages/manager/modules/pci/src/projects/project/data-processing/submit-job/spark-config/spark-config.constants.js (hunk 4 above)
Comment (jperchoc) [reply packages/manager/modules/pci/src/projects/project/data-processing/submit-job/spark-config/spark-config.constants.js:3]: Hi @y4nnL ,
I'd rather keep the constant as an array for two reasons:
- keep consistency with the JAVA_FILES_CONTENT_TYPES constant (same logic)
- we might want to allow more python files types to be selected

File: packages/manager/modules/pci/src/projects/project/data-processing/submit-job/spark-config/spark-config.controller.js (hunk 5 above)
Comment (jperchoc) [reply packages/manager/modules/pci/src/projects/project/data-processing/submit-job/spark-config/spark-config.controller.js:112]: Same as here: https://github.com/ovh/manager/pull/7069#discussion_r926405560

File: packages/manager/modules/pci/src/projects/project/data-processing/submit-job/spark-config/spark-config.controller.js (hunk 6 above)
Comment (jperchoc) [reply packages/manager/modules/pci/src/projects/project/data-processing/submit-job/spark-config/spark-config.controller.js:119]: We only pre-select the first file found, as most of the time, users only have one file. The user can change the selection in the UI if the pre-selection is incorrect though.

File: packages/manager/modules/pci/src/projects/project/data-processing/submit-job/spark-config/spark-config.controller.js (hunk 7 above)
Comment (jperchoc) [reply packages/manager/modules/pci/src/projects/project/data-processing/submit-job/spark-config/spark-config.controller.js:122]: Same as here: https://github.com/ovh/manager/pull/7069#discussion_r926408399

File: packages/manager/modules/pci/src/projects/project/data-processing/submit-job/spark-config/spark-config.constants.js (hunk 4 above)
Comment (y4nnL) [reply packages/manager/modules/pci/src/projects/project/data-processing/submit-job/spark-config/spark-config.constants.js:3]: IMO we should do what logic demands, hence the suggestion. If one day this scenario occurs, let's do those changes ?

Comment (jperchoc) [reply packages/manager/modules/pci/src/projects/project/data-processing/submit-job/spark-config/spark-config.constants.js]: Change done here: https://github.com/ovh/manager/pull/7069/commits/f767af04feeb30add4d8b4cf45c9704320c3aa14

**Review by y4nnL — APPROVED** [review]

**Pull request merged**